### ISSUE TITLE
Missing label on recaptcha form field (accessibility)

### ISSUE THREAD

**Issue by sneelz** [google/recaptcha#421]
**Issue description**

When you run a webpage with the recaptcha on it through accessibility software like WAVE, it calls out a missing form field on the recaptcha. It seems to be on a textarea field that is not actually visible to the user until you turn the page styles off. I'm not sure if this is specific to ng-recaptcha or recaptcha overall.

**Environment**

Using ng-recaptcha to render via Angular UI, but I think this may be an issue with recaptcha itself.

 * OS name and version:
 * PHP version:
 * Web server name and version:
 * `google/recaptcha` version:
 * Browser name and version: 

**Reproducing the issue**

https://dethariel.github.io/ng-recaptcha/basic

  * URL (optional): 
  * Code (optional): 
  
  ***User steps***
  
  
Instructions:
1. Go to any site using recaptcha (e.g. https://dethariel.github.io/ng-recaptcha/basic)
2. Run WAVE on the page: https://chrome.google.com/webstore/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh

Example screenshot of ng-recaptcha WAVE result with styles:

![image](https://user-images.githubusercontent.com/11659471/106039172-ccb23200-60a6-11eb-8d2e-e877a3ca9cf8.png)

Example screenshot of ng-recaptcha WAVE result without styles:

![image](https://user-images.githubusercontent.com/11659471/106039190-d20f7c80-60a6-11eb-9a4c-0c3da27b8837.png)

**Comment by PushEngineering** [comment]
Hi, is there any update on this? Google reCaptcha is still not accessibility compliant

**Comment by bikbal** [comment]
Hi, is there any update on this? Google reCaptcha is still not accessibility compliant.

**Comment by jandrij** [comment]
Issue is still present.

**Comment by webmousegr** [comment]
This textarea appears when a error is encountered, the error triggers because the page is loaded inside Wave web application, in normal circumstances this orphan textarea will not appear, thus the G-CAPTCHA is not always failing the WCAG standard, only when a error is triggered.

If you are interested in passing the WAVE app test for reporting purposes this might help:

`
	<script>
	window.addEventListener('load', function () {
		const label = document.createElement("label");
		const label_text = document.createTextNode("CAPTCHA");
		label.appendChild(label_text);
		label.setAttribute("for", "g-recaptcha-response-100000");
		label.classList.add("screen-reader-text"); // style this class to hide the label (native in wordpress)
		const input_for_label = document.getElementById("g-recaptcha-response-100000");
		input_for_label.insertAdjacentElement("beforebegin", label); 
	})

	</script>
`

in case of wordpress you can wrap this script in functions.php:

`
	add_action('wp_head','g_captcha_wcag_fix_js');

	function g_captcha_wcag_fix_js()
	{ // break out of php ?>
		<script>
		// paste the script here...
		</script>
	<?php } // break back into php
`

**Comment by achilleaspotsis** [comment]
Just run across this error today...
The simplest solution I found was to add somewhere inside the body tag, a label with a for attribute equal to the recaptcha textarea id.
Then just hide it with CSS and you are ready to go!

`<label for="g-recaptcha-response-100000" class="d-none">reCAPTCHA</label>`

**Comment by smustgrave** [comment]
Still experiencing this too.  All of our pages with Google recaptcha are getting pinged for missing form label.

**Comment by jamesmacwhite** [comment]
@smustgrave Yes, but automated tools aren't 100% and they aren't factoring in the display: none; property. All tools that throw an error are likely using the same or similar parsers like HTML_CodeSniffer which will always flag this in a literal sense.

See the note on rules of ARIA (fourth)

https://www.w3.org/TR/using-aria/#fourth

> If an interactive element is hidden using display:none or visibility:hidden (either on the element itself, or any of the element's ancestors), it won't be focusable, and it will also be removed from the [accessibility tree](https://www.w3.org/TR/accname-aam-1.1/#dfn-accessibility-tree). This makes the addition of aria-hidden="true" or explicitly setting tabindex="-1" unnecessary.

While you can apply aria-hidden yourself even the W3C say it is redundant.
